tensorflow/core/lib/strings/str_util_test.cc

```diff
@@ -417,4 +417,33 @@ TEST(StringReplace, EmptyStringReplaceAll) {
   EXPECT_EQ("", str_util::StringReplace("", "a", "X", /*replace_all=*/true));
 }
 
+TEST(SplitUTF8, Basic) {
+  std::vector<string> result;
+  // UTF8 \xE6\x82\xA8 \xE5\xA5\xBD \xE6\x82\xA8 \xE5\xA5\xBD
+  EXPECT_EQ(
+      Status::OK(),
+      str_util::SplitUTF8("\xE6\x82\xA8\xE5\xA5\xBD\xE4\xB8\x96\xE7\x95\x8C",
+                          "", &result));
+  EXPECT_EQ(result.size(), 4);
+
+  // UTF8 \xE6\x82\xA8 \xE5\xA5\xBD \xE4\xB8\x96 \xE7\x95\x8C
+  EXPECT_EQ(
+      Status::OK(),
+      str_util::SplitUTF8("\xE6\x82\xA8\xE5\xA5\xBD\xE4\xB8\x96\xE7\x95\x8C",
+                          "\xE5\xA5\xBD", &result));
+  EXPECT_EQ(result.size(), 2);
+
+  EXPECT_EQ(errors::InvalidArgument("Invalid UTF8 encoding at byte of 1"),
+            str_util::SplitUTF8("\xE2\x28\xA1", "", &result));
+
+  EXPECT_EQ(errors::InvalidArgument("Not enough characters for UTF8 encoding"),
+            str_util::SplitUTF8("\xE6\x82", "", &result));
```

Comment:
Perhaps also include the simpler string tests, like the ones in `Split.Basic`?

Reply:
Thanks. The tests with `Split.Basic` has been added to test against `SplitUTF8` as well.